### DISCUSSION TITLE
鸿蒙在页面上渲染pag文件是用canvas吗？

### DISCUSSION THREAD

**Discussion by zhangruoting** [Tencent/libpag#2268] [General]
鸿蒙在页面上渲染pag文件是用canvas吗？

**Comment by kevingpqi123** [comment]
不是，pag 是跨平台的，渲染都是在 C++ 层，通过自研 2D 渲染引擎 tgfx，平台侧提供 OpenGL 渲染环境

**Reply by zhangruoting** [reply]
那在前端想要展示动效是先调用load,在调用play吗？

**Reply by zhangruoting** [reply]
有没有pag的讨论群啊

**Reply by kevingpqi123** [reply]
后续问题建议将问题描述清楚，如果是 web页面 不就是基于浏览器吗，不知道怎么使用直接看 pag 的 web 平台 demo 工程

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
